bittide/src/Bittide/ProcessingElement.hs

```diff
@@ -40,42 +42,56 @@ processingElement ::
   ( HiddenClockResetEnable dom
   , KnownNat nBusses, 2 <= nBusses, CLog 2 nBusses <= 30) =>
   PeConfig nBusses ->
-  Vec (nBusses-2) (Signal dom (WishboneS2M (Bytes 4))) ->
-  Vec (nBusses-2) (Signal dom (WishboneM2S (32 - CLog 2 nBusses) 4 (Bytes 4)))
-processingElement config bussesIn = case config of
-  PeConfig memMapConfig initI initD ->
-    go memMapConfig initI initD
+  Circuit () (Vec (nBusses-2) (Wishbone dom 'Standard (MappedBus 32 nBusses) (Bytes 4)))
+processingElement (PeConfig memMapConfig' initI' initD') =
+  go memMapConfig' initI' initD'
  where
   go ::
     ( KnownNat depthI, 1 <= depthI
     , KnownNat depthD, 1 <= depthD) =>
     MemoryMap nBusses ->
     InitialContent depthI (Bytes 4) ->
     InitialContent depthD (Bytes 4) ->
-    Vec (nBusses-2) (Signal dom (WishboneM2S (32 - CLog 2 nBusses) 4 (Bytes 4)))
-  go memMapConfig initI initD = bussesOut
+    Circuit () (Vec (nBusses-2) (Wishbone dom 'Standard (MappedBus 32 nBusses) (BitVector 32)))
+  go memMapConfig initI initD = circuit $ do
+    (iBus, dBus) <- rvCircuit (pure low) (pure low) (pure low)
+    ([iMemBus, dMemBus], extBusses) <-
+      (splitAtCircuit d2 <| singleMasterInterconnect memMapConfig) -< dBus
+    wbStorage initI -< iBus
+    wbStorageDPC initD -< (iMemBus, dMemBus)
+    idC -< extBusses
+
+splitAtCircuit ::
+  SNat left ->
+  Circuit (Vec (left + right) a) (Vec left a , Vec right a)
+splitAtCircuit SNat = Circuit go
+ where
+  go (fwd,(bwdLeft, bwdRight)) = (bwd,(fwdLeft, fwdRight))
+   where
+    (fwdLeft, fwdRight) = splitAtI fwd
+    bwd = bwdLeft ++ bwdRight
+
+rvCircuit ::
+  (HiddenClockResetEnable dom) =>
+  Signal dom Bit ->
+  Signal dom Bit ->
+  Signal dom Bit ->
+  Circuit () (Wishbone dom 'Standard 32 (Bytes 4), Wishbone dom 'Standard 32 (Bytes 4))
```

Comment:
More of a philosophical question than a review, but when should we use `Signal dom X -> Circuit () ...` vs `Circuit (CSignal dom X) ...`?

Reply:
That's a great question and the answer is a strong: _I don´t know yet_. Perhaps @martijnbastiaan does.
I can imagine we´d want to rewrite this to:
```hs
rvCircuit ::
  (HiddenClockResetEnable dom) =>
  Circuit 
    (CSignal dom Bit, CSignal dom Bit, CSignal dom Bit) 
    (Wishbone dom 'Standard 32 (Bytes 4), Wishbone dom 'Standard 32 (Bytes 4))
```

Reply:
Using `CSignal` outside of a `Circuit` means that you can't use `|>` to compose anymore, provided that you have another `Circuit` with the right shape. Other than that there is no difference AFAIK. (Perhaps the plugin doesn't understand if you put it outside?) I'd slightly prefer yeeting everything in `Circuit`.

I also think we should use vertical space a bit more aggressively; just treat them as arguments / results, so:

```haskell
  Circuit
    ( CSignal dom Bit
    , CSignal dom Bit
    , CSignal dom Bit
    ) 
    ( Wishbone dom 'Standard 32 (Bytes 4)
    , Wishbone dom 'Standard 32 (Bytes 4)
    )
```

makes it much more obvious how many buses the circuit has. (Alternatively, the trailing `)` can go on the previous line. I don't mind.)

Reply:
I feel that does make it more difficult to partially apply cirucits.

e.g.
With the current implementation you can simply write:
```hs
rv' = rvCircuit (pure 0) (pure 0) (pure 0)
```
to partially apply `rvCircuit`.

Reply:
Makes sense

Reply:
Any idea what the circuit plugin does?